### PR TITLE
Isonline cluster

### PR DESCRIPTION
**Pull request by garrensmith**
Adds functionality to `nmo isonline clusterName` to do a check if all nodes in a cluster are online.

### CODE REVIEW

File: test/isonline.js [hunk 1]
```diff
@@ -138,9 +160,10 @@ lab.experiment('isonline', () => {
         assert.ok(/online/.test(args[1]), 'returns online for online nodes');
         done();
       };
-      nmo.load({nmoconf: __dirname + '/fixtures/randomini'})
+      nmo.load({nmoconf: __dirname + '/fixtures/randomini', json: false})
         .then(() => {
           cli(common.NODE);
+          done();
```
Comment (robertkowalski) [inline test/isonline.js]: with the done()added there this async tests ends too early. if you take a look at the block above, line 159 - where i monkey patch console.log to verify that we are logging the result to the console and then call done - that is the asserting part of the test

i think that is the reason why test coverage went down

File: src/isonline.js [hunk 2]
```diff
@@ -31,12 +35,16 @@ function isOnlineCli (...urls) {
           }
 
           console.log(entry, msg);
-          resolve(results);
+          resolve(msg);
```
Comment (robertkowalski) [inline src/isonline.js]: not sure if you want to change the programmatic api in this place. the intention here is to return json and not a human readable textstring, as it is harder to process from consumers. teh return value of the promise is the programmatic api for http://robertkowalski.github.io/nmo/api/nmo-isonline.html

Comment (garrensmith) [inline src/isonline.js]: Good spot. That was a bad typo on my part.

**Comment by robertkowalski** [comment]
if you ask isonline to see if a non existing cluster is online it crashes, +1 after checking for it with an error of error.type `EUSAGE`

```
(17:23:07) [robert@tequila-work] ~/apache/couchdb-nmo (pr/2) $ ./bin/nmo-cli.js isonline notexist
ERR! Object.keys called on non-object
ERR! TypeError: Object.keys called on non-object
ERR!     at Function.keys (native)
ERR!     at getClusterUrls (/Users/robert/apache/couchdb-nmo/lib/isonline.js:76:17)
ERR!     at /Users/robert/apache/couchdb-nmo/lib/isonline.js:50:14
ERR!     at tryCatcher (/Users/robert/apache/couchdb-nmo/node_modules/bluebird/js/main/util.js:24:31)
ERR!     at Promise._resolveFromResolver (/Users/robert/apache/couchdb-nmo/node_modules/bluebird/js/main/promise.js:427:31)
ERR!     at new Promise (/Users/robert/apache/couchdb-nmo/node_modules/bluebird/js/main/promise.js:53:37)
ERR!     at isOnlineCli (/Users/robert/apache/couchdb-nmo/lib/isonline.js:41:10)
ERR!     at /Users/robert/apache/couchdb-nmo/bin/nmo-cli.js:34:6
ERR!     at tryCatcher (/Users/robert/apache/couchdb-nmo/node_modules/bluebird/js/main/util.js:24:31)
ERR!     at Promise._settlePromiseFromHandler (/Users/robert/apache/couchdb-nmo/node_modules/bluebird/js/main/promise.js:454:31)
ERR!     at Promise._settlePromiseAt (/Users/robert/apache/couchdb-nmo/node_modules/bluebird/js/main/promise.js:530:18)
ERR!     at Promise._settlePromises (/Users/robert/apache/couchdb-nmo/node_modules/bluebird/js/main/promise.js:646:14)
ERR!     at Async._drainQueue (/Users/robert/apache/couchdb-nmo/node_modules/bluebird/js/main/async.js:177:16)
ERR!     at Async._drainQueues (/Users/robert/apache/couchdb-nmo/node_modules/bluebird/js/main/async.js:187:10)
ERR!     at Immediate.Async.drainQueues [as _onImmediate] (/Users/robert/apache/couchdb-nmo/node_modules/bluebird/js/main/async.js:15:14)
ERR!     at processImmediate [as _immediateCallback] (timers.js:358:17)
ERR!
ERR!
ERR! nmo: 1.0.1 node: v0.12.2
ERR! please open an issue including this log on https://github.com/robertkowalski/nmo/issues
```

File: src/isonline.js [hunk 3]
```diff
@@ -37,6 +41,13 @@ function isOnlineCli (...urls) {
   });
 }
 
+export function getClusterUrls (clusterName) {
+  const nodes = nmo.config.get(clusterName);
+  if (!nodes) {
+    throw new Error('Cluster does not exist');
```
Comment (robertkowalski) [inline src/isonline.js]: can you set the error type to EUSAGE?

using EUSAGE as error type hides the long stacktrace and the message that the user should submit an issue in the bugtracker

example: https://github.com/garrensmith/couchdb-nmo/blob/isonline-cluster/src/isonline.js#L14

code handling error types: https://github.com/garrensmith/couchdb-nmo/blob/isonline-cluster/bin/nmo-cli.js#L39-L59

**Comment by robertkowalski** [comment]
+1 after setting the error type to `EUSAGE`

**Comment by garrensmith** [comment]
Merged!

**Pull request closed without merging**